doc/users/next_whats_new/sides_violinplot.rst

```diff
@@ -0,0 +1,4 @@
+Add option to plot only one half of violin plot
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Setting the parameter `side` to 'left' or 'right' allows to only plot one half of the violin plot.
```

Comment:
In https://github.com/matplotlib/matplotlib/issues/27812#issuecomment-1960450463 I suggested values "low" / "high" / "both", but I'm a bit torn myself. Left/right is slightly better (because simpler) for vertical violins. However, the function also supports horizontal bars, where left/right is quite difficult to interpret.

Reply:
Ah sorry, no I forgot you suggested that. It's much better than left/right. I'll change it.

Reply:
On a similar note to the `capstyle="projecting"` comment: I think it would be nice to move the vertical lines by 0.5*linewidth away from the center so that both are shown and not just the one that's plotted last.
Is there a way to translate the linewidth to an absolute unit or is this impossible?

Reply:
This is not consistently possible. While you could calculate the coordinate shift for a certain state, that would break as soon as you change axes limits/zoom or dpi. The correct way to make this happen would be half-sided lines, but we don’t have that concept. I therefore would refrain from adjustments in the violinplot function.

Reply:
Ok that's what I suspected.